### PR TITLE
P4CP : DefaultActionSupport

### PR DESCRIPTION
**Pull request by aashishkuma**
This PR contains the changes required in the p4rt-ctl to support default entry/action modification done during runtime.
The supported operations are:
set-default-entry : 
- to either set the default entry to a valid action for the table
- to either reset the default entry to its original value as defined in the P4 program.

get-default-entry
- to fetch the default entry for a table

Example:
**Set Operation**: $P4CP_INSTALL/bin/p4rt-ctl set-default-entry br0 linux_networking_control.source_port_to_bridge_map "linux_networking_control.set_bridge_id(bridge_id=10)"
**Reset Operation**: $P4CP_INSTALL/bin/p4rt-ctl set-default-entry br0 linux_networking_control.source_port_to_bridge_map ""
**Get Operation**: $P4CP_INSTALL/bin/p4rt-ctl get-default-entry br0 linux_networking_control.source_port_to_bridge_map

P4Runtime Spec for Default Entry:
The default entry is identified by setting the is_default_action boolean field to true. When this flag is set to true, the repeated match field must be empty and the priority field must be set to zero, otherwise the P4Runtime server must return an INVALID_ARGUMENT error code. When performing a MODIFY update on the default entry, the client can either provide a valid action for the table or leave the action field unset, in which case the default entry will be reset to its original value, as defined in the P4 program.

### CODE REVIEW

File: clients/p4rt-ctl/p4rt-ctl.in [hunk 1]
```diff
@@ -467,24 +467,25 @@ class P4InfoHelper(object):
...
+                # A table can contain more than one match filed and in such cases
+                # if one match_type has ternary and other filed is of any type
+                # it is considered as ternary match_type.
+                # In these cases we expect priority field from the user.
+                p4info_match = self.get_match_field(table_name, mfn)
+
+                if p4info_match.match_type == p4info_pb2.MatchField.TERNARY:
+                    one_match_type_is_ternary = True
+                    # If match_type is ternary, then priority field is expected
+                    if priority == None:
```
Comment (5abeel) [inline clients/p4rt-ctl/p4rt-ctl.in:481]: `priority is None` ?

In Python, the recommendation is to use `is` instead of `==` when comparing to `None`

**Review by 5abeel — APPROVED** [review]
lgtm

**Review by ffoulkes — APPROVED** [review]
LGTM

File: clients/p4rt-ctl/p4rt-ctl.in [hunk 2]
```diff
@@ -467,24 +467,25 @@ class P4InfoHelper(object):
...
-                    print("For ternary match_type, priority field is needed")
-                    sys.exit(1)
-
-            if match_fields:
-                table_entry.match.extend([
-                    self.get_match_field_pb(table_name, mfn, value)
-                ])
+        if match_fields:
+            for mfn, value in match_fields.items():
+                # A table can contain more than one match filed and in such cases
```
Comment (ffoulkes) [inline clients/p4rt-ctl/p4rt-ctl.in:472]:
```suggestion
                # A table can contain more than one match field, and in such cases,
```

File: clients/p4rt-ctl/p4rt-ctl.in [hunk 3]
```diff
@@ -467,24 +467,25 @@ class P4InfoHelper(object):
...
+                # if one match_type has ternary and other filed is of any type
```
Comment (ffoulkes) [inline clients/p4rt-ctl/p4rt-ctl.in:473]:
```suggestion
                # if one match_type has ternary and another field is of any type,
```

File: clients/p4rt-ctl/p4rt-ctl.in [hunk 4]
```diff
@@ -467,24 +467,25 @@ class P4InfoHelper(object):
...
+                # it is considered as ternary match_type.
+                # In these cases we expect priority field from the user.
```
Comment (ffoulkes) [inline clients/p4rt-ctl/p4rt-ctl.in:475]:
```suggestion
                # In these cases, we expect a priority field from the user.
```

**Review by ffoulkes — COMMENTED** [review]
Note spelling and punctuation corrections.

**Pull request merged**